starknet_py/net/schemas/rpc/transactions.py

```diff
@@ -0,0 +1,420 @@
+    }
+
+    def get_obj_type(self, obj):
+        return _extract_tx_version(obj.version)
+
+    def get_data_type(self, data):
+        return _extract_tx_version(data.get("version"))
+
+
+class DeployAccountTransactionSchema(OneOfSchema):
+    type_schemas = {
+        "1": DeployAccountTransactionV1Schema,
+        "3": DeployAccountTransactionV3Schema,
+    }
+
+    def get_obj_type(self, obj):
+        return _extract_tx_version(obj.version)
+
+    def get_data_type(self, data):
+        return _extract_tx_version(data.get("version"))
+
+
+class L1HandlerTransactionSchema(TransactionSchema):
+    contract_address = Felt(data_key="contract_address", required=True)
+    calldata = fields.List(Felt(), data_key="calldata", required=True)
+    entry_point_selector = Felt(data_key="entry_point_selector", required=True)
+    nonce = NumberAsHex(data_key="nonce", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> L1HandlerTransaction:
+        return L1HandlerTransaction(**data)
+
+
+class TypesOfTransactionsSchema(OneOfSchema):
+    type_field = "type"
+    type_schemas = {
+        "INVOKE": InvokeTransactionSchema,
+        "DECLARE": DeclareTransactionSchema,
+        "DEPLOY": DeployTransactionSchema,
+        "DEPLOY_ACCOUNT": DeployAccountTransactionSchema,
+        "L1_HANDLER": L1HandlerTransactionSchema,
+    }
+
+
+class StarknetBlockWithTxHashesSchema(BlockHeaderSchema):
+    status = BlockStatusField(data_key="status", required=True)
+    transactions = fields.List(Felt(), data_key="transactions", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> StarknetBlockWithTxHashes:
+        return StarknetBlockWithTxHashes(**data)
+
+
+class TransactionWithReceiptSchema(Schema):
+    transaction = fields.Nested(TypesOfTransactionsSchema(), data_key="transaction")
+    receipt = fields.Nested(TransactionReceiptSchema(), data_key="receipt")
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> TransactionWithReceipt:
+        return TransactionWithReceipt(**data)
+
+
+class SentTransactionSchema(Schema):
+    transaction_hash = Felt(data_key="transaction_hash", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> SentTransactionResponse:
+        return SentTransactionResponse(**data)
+
+
+class DeclareTransactionResponseSchema(SentTransactionSchema):
+    class_hash = Felt(data_key="class_hash", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> DeclareTransactionResponse:
+        return DeclareTransactionResponse(**data)
+
+
+class DeployAccountTransactionResponseSchema(SentTransactionSchema):
+    address = Felt(data_key="contract_address", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> DeployAccountTransactionResponse:
+        return DeployAccountTransactionResponse(**data)
+
+
+class PendingStarknetBlockWithReceiptsSchema(PendingBlockHeaderSchema):
```

Comment:
Why are the `Block` schemas from this file not in `rpc/block.py` instead?

Reply:
These schemas have been in the `transactions.py` because of their dependency with transaction schemas in this file, moved to `block.py` cause circular import error.

Reply:
Please try to fix this. If we move all block schemas to `block.py`, we will have a problem where we depend on `ExecutionResourcesSchema` within `transactions.py`. We could consider creating an additional file, perhaps `general.py` into which we can move elements like `ExecutionResourcesSchema` and `EstimatedFeeSchema`.